Mathlib/Data/Fin/Tuple/Basic.lean

```diff
@@ -623,12 +639,46 @@ theorem append_right_eq_snoc {α : Type*} {n : ℕ} (x : Fin n → α) (x₀ : F
     exact (@snoc_last _ (fun _ => α) _ _).symm
 #align fin.append_right_eq_snoc Fin.append_right_eq_snoc
 
+/-- `Fin.snoc` is the same as appending a one-tuple -/
+theorem snoc_eq_append {α : Type*} (xs : Fin n → α) (x : α) :
+    snoc xs x = append xs (cons x Fin.elim0) :=
+  (append_right_eq_snoc xs (cons x Fin.elim0)).symm
```

Comment:
Do we have this lemma for `cons` too?

Reply:
We have the counterpart of `append_right_eq_snoc`, but not phrased in a way to rewrite arbitrary `cons` applications. I've now added:
```lean
theorem cons_eq_append {α : Type*} (x : α) (xs : Fin n → α) :
    cons x xs = append (cons x Fin.elim0) xs ∘ Fin.cast (add_comm ..)
```